utils/pinwheels.py

```diff
@@ -0,0 +1,98 @@
+import pathlib
+import matplotlib.pyplot as plt
+import numpy as np
+import os 
+
+def plot_pinwheel(sample_id, final_scores, id, name):
+    """
+    Generates and saves a polar plot (pinwheel) visualization for pathway scores.
+
+    Parameters:
+        sample_id (str): Identifier for the sample being visualized.
+        final_scores (DataFrame): DataFrame containing pathway IDs and scores.
+        id (str): Column name for pathway identifiers.
+        name (str): Type of plot - e.g. pathway or drug.
+    """
+    save_dir = pathlib.Path("./visualize/pinwheels").resolve()
```

Comment:
consider ensuring that directory exists before executing the whole function. 
```suggestion
    save_dir = pathlib.Path("./visualize/pinwheels").resolve()
    save_dir.mkdir(parents=True, exists_ok=True)
```

Reply:
done